compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/service/validator/ServiceValidator.java

```diff
@@ -800,12 +806,12 @@ private void validateRecordFields(RecordTypeSymbol recordTypeSymbol, Location lo
     }
 
     private void addDiagnostic(CompilationDiagnostic compilationDiagnostic, Location location) {
-        this.errorOccurred = true;
+        this.errorOccurred = compilationDiagnostic.getDiagnosticSeverity() == DiagnosticSeverity.ERROR;
```

Comment:
Is this logic correct ? what happens if there was an error added previously and the final call added with a warning ?
I think this should be 
```
this.errorOccurred = compilationDiagnostic.getDiagnosticSeverity() == DiagnosticSeverity.ERROR ? true : this.errorOccurred;

```

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-graphql/pull/1518/commits/6c7e8043decd4f264b6fb5a4338067be8a18e6de